aggregator/aggregator.go

```diff
@@ -43,275 +40,238 @@ func NewAggregator(
-				log.Error("inputHash received from the prover (", publicInputsExtended.InputHash,
-					") doesn't match with the internal value: ", internalInputHashS)
-				log.Debug("internalBatchHashData: ", batchHashData, " externalBatchHashData: ", publicInputsExtended.PublicInputs.BatchHashData)
-				log.Debug("inputProver.PublicInputs.OldStateRoot: ", inputProver.PublicInputs.OldStateRoot)
-				log.Debug("inputProver.PublicInputs.OldLocalExitRoot:", inputProver.PublicInputs.OldLocalExitRoot)
-				log.Debug("inputProver.PublicInputs.NewStateRoot: ", inputProver.PublicInputs.NewStateRoot)
-				log.Debug("inputProver.PublicInputs.NewLocalExitRoot: ", inputProver.PublicInputs.NewLocalExitRoot)
-				log.Debug("inputProver.PublicInputs.SequencerAddr: ", inputProver.PublicInputs.SequencerAddr)
-				log.Debug("inputProver.PublicInputs.BatchHashData: ", inputProver.PublicInputs.BatchHashData)
-				log.Debug("inputProver.PublicInputs.BatchNum: ", inputProver.PublicInputs.BatchNum)
-				log.Debug("inputProver.PublicInputs.EthTimestamp: ", inputProver.PublicInputs.EthTimestamp)
-			}
-
-			// 4. send proof + txs to the SC
-			err = a.EthTxManager.VerifyBatch(batchToVerify.BatchNumber, resGetProof)
-			if err != nil {
-				log.Warnf("failed to send request to consolidate batch to ethereum, batch number: %d, err: %v",
-					batchToVerify.BatchNumber, err)
-				continue
-			}
-			batchesSent[batchToVerify.BatchNumber] = true
-		case <-a.ctx.Done():
-			getProofCtxCancel()
-			return
+		a.tryVerifyBatch(ctx, ticker)
+	}
+}
+
+func (a *Aggregator) tryVerifyBatch(ctx context.Context, ticker *time.Ticker) {
+	log.Info("checking if network is synced")
+	for !a.isSynced(ctx) {
+		log.Infof("waiting for synchronizer to sync...")
+		waitTick(ctx, ticker)
+		continue
+	}
+	log.Info("network is synced, getting batch to verify")
+	batchToVerify, err := a.getBatchToVerify(ctx)
+	if err != nil {
+		waitTick(ctx, ticker)
+		return
+	}
+	log.Infof("checking profitability to aggregate batch, batchNumber: %d", batchToVerify.BatchNumber)
+	// pass matic collateral as zero here, bcs in smart contract fee for aggregator is not defined yet
+	isProfitable, err := a.ProfitabilityChecker.IsProfitable(ctx, big.NewInt(0))
+	if err != nil {
+		log.Warnf("failed to check aggregator profitability, err: %v", err)
+		return
+	}
+
+	if !isProfitable {
+		log.Infof("Batch %d is not profitable, matic collateral %d", batchToVerify.BatchNumber, big.NewInt(0))
+		return
+	}
+
+	log.Infof("sending zki + batch to the prover, batchNumber: %d", batchToVerify.BatchNumber)
+	inputProver, err := a.buildInputProver(ctx, batchToVerify)
+	if err != nil {
+		log.Warnf("failed to build input prover, err: %v", err)
+		return
+	}
+
+	genProofID, err := a.ProverClient.GetGenProofID(ctx, inputProver)
+	if err != nil {
+		log.Warnf("failed to get gen proof id, err: %v", err)
+		return
+	}
+
+	resGetProof, err := a.ProverClient.GetResGetProof(ctx, genProofID, batchToVerify.BatchNumber)
```

Comment:
shouldn't this be in a loop waiting for the proof generation to either complete (which takes a considerable amount of time) or fail?

Reply:
Yes, it's in the loop, it's here - https://github.com/0xPolygonHermez/zkevm-node/pull/964/files#diff-1a6a23e9ac05b8b9b16f35d7459375093c07b45bfb35da0fe9799514d8e3cc97R54